### PR TITLE
 Add kind inheritance related to CONTROL_FLOW and JUMP

### PR DESCRIPTION
**Pull request by alban-auzeill**

### CODE REVIEW

File: common-rule-engine/src/main/java/org/sonar/uast/UastNode.java [hunk 1]
```diff
@@ -101,6 +102,9 @@ public String toString() {
   }
 
   public enum Kind implements Predicate<UastNode> {
+    // TODO when WHILE and TRY will be added, they will have to extends from CONTROL_FLOW
```
Comment (SonarTech) [inline common-rule-engine/src/main/java/org/sonar/uast/UastNode.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.go%3Asonar-go&pullRequest=223&issues=AWKz9dccYna8yCuHfvaB&open=AWKz9dccYna8yCuHfvaB)

[ISSUE_KEY]: AWKz9dccYna8yCuHfvaB

File: common-rule-engine/src/main/java/org/sonar/uast/UastNode.java [hunk 2]
```diff
@@ -101,6 +102,9 @@ public String toString() {
   }
 
   public enum Kind implements Predicate<UastNode> {
+    // TODO when WHILE and TRY will be added, they will have to extends from CONDITIONAL_JUMP
```
Comment (SonarTech) [inline common-rule-engine/src/main/java/org/sonar/uast/UastNode.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.go%3Asonar-go&pullRequest=223&issues=AWKz9dccYna8yCuHfvaB&open=AWKz9dccYna8yCuHfvaB)

[ISSUE_KEY]: AWKz9dccYna8yCuHfvaB

**Review by saberduck — APPROVED** [review]

**Review by SonarTech — COMMENTED** [review]
SonarQube analysis found issues:
Bug Bugs: 0
Vulnerability Vulnerabilities: 0
Code Smell Code Smells: 1

[See all issues in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.go%3Asonar-go&pullRequest=223&resolved=false)

File: common-rule-engine/src/main/java/org/sonar/uast/UastNode.java (hunk 2 above)
Comment (SonarTech) [inline common-rule-engine/src/main/java/org/sonar/uast/UastNode.java:105]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.go%3Asonar-go&pullRequest=223&issues=AWKz9dccYna8yCuHfvaB&open=AWKz9dccYna8yCuHfvaB)

[ISSUE_KEY]: AWKz9dccYna8yCuHfvaB

**Pull request merged**